build-system/babel-plugins/babel-plugin-mangle-string-pattern/index.js

```diff
@@ -0,0 +1,111 @@
+const dedent = require('dedent');
+const {basename, dirname} = require('path');
+const {outputJsonSync, readJsonSync} = require('fs-extra');
+
+// Use our own charset instead of base62 since HTML classnames are case insensitive
+const CHARSET = '0123456789abcdefghijklmnopqrstuvwxyz-'.split('');
+
+/**
+ * @param {number} int
+ * @return {string}
+ */
+function encode(int) {
+  if (int === 0) {
+    return CHARSET[0];
+  }
+  const {length} = CHARSET;
+  let res = '';
+  while (int > 0) {
+    res = CHARSET[int % length] + res;
+    int = Math.floor(int / length);
+  }
+  return res;
+}
+
+const pattern = /i-amphtml-story-[a-zA-Z0-9-]*[a-zA-Z0-9]/g;
+const prefix = 'i-amphtml--';
+
+// TODO(alanorozco): These should be prefixed on source. They're a ~250 B delta.
+const specificPattern = new RegExp(
+  dedent(`
+    i-amphtml-tooltip-action-icon-launch
```

Comment:
I think we can take this way farther, essentially everything that matches `i-amphtml-[a-zA-Z0-9_-]+` can be mangled. There are a couple of special cases that are trying to compute the classname we should fix:

```
$ rg --only-matching "\bi-amphtml-[a-zA-Z0-9_-]+(['\"] \+|\\$\{)"
src/shadow-embed.js
73:i-amphtml-sd-${

src/core/dom/layout/index.js
103:i-amphtml-layout-' +

extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-debug.js
50:i-amphtml-story-dev-tools-log-status-icon-' +

extensions/amp-story-dev-tools/0.1/amp-story-dev-tools-tab-preview.js
736:i-amphtml-story-dev-tools-device-dialog-${

extensions/amp-onetap-google/0.1/test/test-amp-onetap-google.js
330:i-amphtml-onetap-google-ui-${

extensions/amp-onetap-google/0.1/amp-onetap-google.js
176:i-amphtml-onetap-google-ui-${

extensions/amp-form/0.1/form-validators.js
203:i-amphtml-validation-bubble-${

extensions/amp-story/1.0/test/test-amp-story-embedded-component.js
132:i-amphtml-story-focused-state-layer-nav-button' +
153:i-amphtml-story-focused-state-layer-nav-button' +

extensions/amp-story/1.0/semantic-render.js
12:i-amphtml-story-${

extensions/amp-auto-lightbox/0.1/amp-auto-lightbox.js
406:i-amphtml-auto-lightbox-${

extensions/amp-story-interactive/0.1/interactive-disclaimer.js
85:i-amphtml-story-disclaimer-${

extensions/amp-story-auto-ads/0.1/story-ad-page.js
84:i-amphtml-ad-page-${
339:i-amphtml-demo-${

validator/js/engine/validator.js
3806:i-amphtml-layout-' +
```

Reply:
validation should probably exclude both if there is a mangling. so far it's been `-` AFAIK.

Reply:
@jridgewell Some of these dynamic uses are legitimate, like generating unique ids. If we'd like to 

@honeybadgerdontcare `i-amphtml-` is already a forbidden prefix that this transform preserves. No validation changes should be required.

<img width="868" alt="Screen Shot 2021-11-03 at 8 36 35 AM" src="https://user-images.githubusercontent.com/254946/140092710-31be86f4-e102-466d-a9d4-74944bc21339.png">

Reply:
@jridgewell The value of mangling the rest of the source is also negligible. [On `v0.mjs`:](https://gist.github.com/alanorozco/f09d17f36ebaded2c0f77e781eeab760)
```
   62360 main
   62316 mangle-string-pattern
```
`amp-story` benefits particularly because it's way more CSS-heavy than any other bundle.